rts/idris_utf8.c

```diff
@@ -78,6 +78,36 @@ unsigned idris_utf8_index(char* s, int idx) {
    return top;
 }
 
+char* idris_utf8_advance(char* str, int i) {
```

Comment:
> Isnt there the option to get the byte length of the buffer from the idris side?

The length isn't stored with the string so it would just do a strlen, I think my code snippet is better than checking the length anyway (because a stride might pass the length and then you have to get the logic just right to avoid accessing unallocated memory).

> Sorry for the duplication.

I thought it wasn't such a big deal (it works right on correct string), but your warning got me thinking, so thanks for setting me straight.